reconcile/utils/secret_reader.py

```diff
@@ -32,6 +38,16 @@ def vault_client(self):
             self._vault_client = VaultClient()
         return self._vault_client
 
+    @property
+    def vault_enabled(self):
+        if not self.settings:
+            return False
+
+        if isinstance(self.settings, Mapping):
+            return self.settings.get("vault")
```

Comment:
If settings is a mapping, but doesn't have a `vault` key, then this could return `None`? Is that expected or if not we could change this to `get("vault", False)`?

Reply:
good catch!!!